### PR TITLE
Dev nzr

### PR DESCRIPTION
**Pull request by nsmorzhevskyi0024**

### CODE REVIEW

File: Application/Application.csproj [hunk 1]
```diff
@@ -4,4 +4,19 @@
     <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Folder Include="Service Interfaces" />
```
Comment (ArtemGontar) [inline Application/Application.csproj]: Just Interfaces or Abstractions

File: Application/Application.csproj [hunk 2]
```diff
@@ -4,4 +4,19 @@
...
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Domain\Domain.csproj" />
+    <ProjectReference Include="..\Persistence\Persistence.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="MediatR" Version="9.0.0" />
+    <PackageReference Include="MediatR.Extensions.Microsoft.DependencyInjection" Version="9.0.0" />
```
Comment (ArtemGontar) [inline Application/Application.csproj]: No need DependencyInjection here, I suppose

File: Application/Command Handlers/DeleteUserCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        private readonly IUserRepository _userRepository;
+
+        public DeleteUserCommandHandler(IUserRepository userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        public async Task<bool> Handle(DeleteUserCommand command, CancellationToken cancellationToken)
+        {
+            var user = _userRepository.GetUserByIdAsync(command.Id);
```
Comment (ArtemGontar) [inline Application/Command Handlers/DeleteUserCommandHandler.cs]: lack of await keyword

File: Application/Command Handlers/DeleteUserCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                
+            if (await user == null) return false;
```
Comment (ArtemGontar) [inline Application/Command Handlers/DeleteUserCommandHandler.cs]: remove await here, better to write it beyond

File: Application/Command Handlers/UpdateUserCommandHandler.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        private readonly IUserRepository _userRepository;
+
+        public UpdateUserCommandHandler(IUserRepository userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        public async Task<ApplicationUser> Handle(UpdateUserCommand command, CancellationToken cancellationToken)
+        {
+            var userToUpdate = _userRepository
```
Comment (ArtemGontar) [inline Application/Command Handlers/UpdateUserCommandHandler.cs]: add await here

File: Application/Dependency Injection/DependencyInjection.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Application.Dependency_Injection
+{
+    public static class DependencyInjection
+    {
+        public static void AddApplication(this IServiceCollection services)
+        {
+            services.AddMediatR(Assembly.GetExecutingAssembly());
```
Comment (ArtemGontar) [inline Application/Dependency Injection/DependencyInjection.cs]: I think this class is superfluous. Better just addMediatR in startup

File: Application/Dependency Injection/DependencyInjection.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Reflection;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Application.Dependency_Injection
```
Comment (ArtemGontar) [inline Application/Dependency Injection/DependencyInjection.cs]: You can set namespace 'Microsoft.Extensions.DependencyInjection;
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-5.0
Register groups of services with extension methods chapter

File: Application/Query Handlers/GetUserByIdQueryHandler.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        {
+            _userRepository = userRepository;
+        }
+
+        public async Task<ApplicationUser> Handle(GetUserByIdQuery request,
+            CancellationToken cancellationToken)
+        {
+            var user = _userRepository.GetUserByIdAsync(request.Id);
+                
+            if (user == null) return null;
```
Comment (ArtemGontar) [inline Application/Query Handlers/GetUserByIdQueryHandler.cs:24]: Please check your code, you try to check nullable for Task object. IT IS SUPER USELESS
Just add await for 22 line of code

File: Application/Query Handlers/GetUsersQueryHandler.cs [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        {
+            _userRepository = userRepository;
+        }
+
+        public async Task<IEnumerable<ApplicationUser>> Handle(GetAllUsersQuery request,
+            CancellationToken cancellationToken)
+        {
+            var users = _userRepository.GetUsersAsync();
+                
+            if (users == null) return null;
```
Comment (ArtemGontar) [inline Application/Query Handlers/GetUsersQueryHandler.cs]: Please check your code, you try to check nullable for Task object. IT IS SUPER USELESS
Just add await for 23 line of code

File: Domain/Domain.csproj [hunk 10]
```diff
@@ -4,4 +4,13 @@
     <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="3.1.9" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="Entity Interfaces" />
```
Comment (ArtemGontar) [inline Domain/Domain.csproj]: Remember - never use white space for naming folders, you can run into issues, and that issues will troubleshoot difficultly

File: Persistence/Contexts/BlogContext.cs [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+        public BlogContext(DbContextOptions<BlogContext> options) : base(options)
+        {
+            Database.EnsureCreated();
+        }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer(
+                "Server=sonnov;Database=BlogIdentityDatabase;Trusted_Connection=true;"
```
Comment (ArtemGontar) [inline Persistence/Contexts/BlogContext.cs]: Replace connectionString to appsettings.json
Documentation provided - https://docs.microsoft.com/en-us/ef/core/miscellaneous/connection-strings

File: Persistence/Persistence.csproj [hunk 12]
```diff
@@ -4,4 +4,22 @@
...
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Identity" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="3.1.9" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="3.1.9" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="3.1.9" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="3.1.9" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.1.9" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="3.1.9" />
```
Comment (ArtemGontar) [inline Persistence/Persistence.csproj]: Are you really need so many EF nugets?
Looks strange

File: WebAPI/Startup.cs [hunk 13]
```diff
@@ -1,34 +1,35 @@
...
-
-        // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddApplication();
             services.AddControllers();
+            services.AddScoped<IUserRepository, UserRepository>();
+            services.AddDbContext<BlogContext>(options => options
+                .UseSqlServer(Configuration.GetConnectionString("BlogConnection"))
+                .UseLazyLoadingProxies());
```
Comment (ArtemGontar) [inline WebAPI/Startup.cs]: Which reason you have to use LazyLoadings?

**Review by ArtemGontar — CHANGES_REQUESTED** [review]

File: Application/Application.csproj (hunk 1 above)
Comment (nsmorzhevskyi0024) [reply Application/Application.csproj]: Done

File: Application/Application.csproj (hunk 2 above)
Comment (nsmorzhevskyi0024) [reply Application/Application.csproj]: Done

File: Application/Command Handlers/DeleteUserCommandHandler.cs (hunk 3 above)
Comment (nsmorzhevskyi0024) [reply Application/Command Handlers/DeleteUserCommandHandler.cs]: Done

File: Application/Command Handlers/DeleteUserCommandHandler.cs (hunk 4 above)
Comment (nsmorzhevskyi0024) [reply Application/Command Handlers/DeleteUserCommandHandler.cs]: Done

File: Application/Command Handlers/UpdateUserCommandHandler.cs (hunk 5 above)
Comment (nsmorzhevskyi0024) [reply Application/Command Handlers/UpdateUserCommandHandler.cs]: Done

File: Domain/Domain.csproj (hunk 10 above)
Comment (nsmorzhevskyi0024) [reply Domain/Domain.csproj]: Got it. Resolved.

File: Application/Query Handlers/GetUsersQueryHandler.cs (hunk 9 above)
Comment (nsmorzhevskyi0024) [reply Application/Query Handlers/GetUsersQueryHandler.cs]: Done

File: WebAPI/Startup.cs (hunk 13 above)
Comment (nsmorzhevskyi0024) [reply WebAPI/Startup.cs]: I have confused about some things. There is really no need =)

File: Application/Dependency Injection/DependencyInjection.cs (hunk 6 above)
Comment (nsmorzhevskyi0024) [reply Application/Dependency Injection/DependencyInjection.cs]: Resolved

File: Application/Query Handlers/GetUserByIdQueryHandler.cs (hunk 8 above)
Comment (nsmorzhevskyi0024) [reply Application/Query Handlers/GetUserByIdQueryHandler.cs:24]: Resolved

**Pull request merged**